### DISCUSSION TITLE
Overlay colors with alpha when annotating

### DISCUSSION THREAD

**Discussion by Uregek** [victorsoares96/epubjs-react-native#289] [Q&A]
I started making my own small reader for personal use, I want to read with a friend and synchronize our annotations, but I ran into a problem that stopped me. When adding an annotation in the standard way presented in the example from the documentation, the colors with transparency overlap each other and in some places a different shade is obtained. 

I roughly understand what this might be connected with. <p> tags are used there, and sometimes the same piece of text can be in nested <p> tags, for example: p>p>text. And the color fill extends to both elements, which is why the overlap occurs. 

Is it possible to solve this problem? I can't move with this. I am attaching two screenshots. The first one is my overlay application. The second is an annotation in the standard "Book" application on the iPhone. I will be very happy for any answer please!

My app: it is all one annotate
<img width="448" alt="Screenshot 2024-07-17 at 1 48 09 AM" src="https://github.com/user-attachments/assets/f6a2389b-be26-496f-8a8b-0311dfc22b46">

"Book" ios app:
![IMG_6906](https://github.com/user-attachments/assets/59367f62-ecb9-4a59-925a-1eb57e9a14a4)

pice of code that i used in: 

```JSX
          menuItems={[
            {
              label: '🟡',
              action: (cfiRange) => {
                addAnnotation('highlight', cfiRange, undefined, {
                  color: '#ffaf02',
                });
                return true;
              },
            },
```
so it is from standard example

**Comment by victorsoares96** [comment]
unfortunately this is not possible at the moment, text markings are basically svg elements over the position(x,y) of the text

**Comment by Uregek** [comment]
I managed to fix the error a little bit by updating the epubjs library, however, it does not work 100%, but it is much better than it was before. 
I am attaching screenshots below. 
To do this, you need to replace the src/epub js file.ts to the new one from the attachment

[epubjs.ts.zip](https://github.com/user-attachments/files/16745463/epubjs.ts.zip)

![IMG_7426](https://github.com/user-attachments/assets/606310a6-ba53-44d0-87c9-5feb811fac4a) 
![IMG_7427](https://github.com/user-attachments/assets/dc020fd4-7255-46ba-aca8-33f85f5d85db)

Unfortunately, I am a pedant, and this solution does not cover 100 percent of cases, so I abandoned my project

**Reply by victorsoares96** [reply]
Did you make this correction in the epubjs file using which repository as a base?

**Reply by Uregek** [reply]
@victorsoares96 I tried with the official version that hangs in npm (build a repo min file), it didn't work, then I found epubjs on the Internet with an even larger version, it was not a repo, it was a cdn like link

**Reply by Cherevatin** [reply]
@Uregek Hi, could you please specify the places where you made changes? I would really appreciate it, as it's quite difficult to find the modifications in the minified code without knowing the source code version. Thank you!

**Reply by Uregek** [reply]
@Cherevatin As I said in the comment above, I did not make my changes to the epubjs repository, I just took a minified file from the Internet with a different version

**Reply by Cherevatin** [reply]
@Uregek Aaaah, I get it. Thanks for that too!

**Comment by victorsoares96** [comment]
very good, thanks for the contribution @Uregek I will implement this in the next version if everything goes well
